codalab/lib/codalab_manager.py

```diff
@@ -497,6 +517,69 @@ def client(self, address, is_cli=True):
     def cli_verbose(self):
         return self.config.get('cli', {}).get('verbose')
 
+    def get_access_token(self, address, auth_handler):
```

Comment:
The main barrier to solving this was that the auth handler needed the address of the REST server (since that's where the oauth endpoints are defined) in order to contact the server directly for authentication (rather than through this ugly RPC call). But maybe we can use the same solutions we just discussed on Slack?